src/main/java/me/gnat008/perworldinventory/data/metadata/PWIMetaDataManager.java

```diff
@@ -0,0 +1,37 @@
+package me.gnat008.perworldinventory.data.metadata;
+
+import ch.jalu.injector.Injector;
+
+import me.gnat008.perworldinventory.PerWorldInventory;
+
+import org.bukkit.metadata.MetadataValue;
+import org.bukkit.entity.Player;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class PWIMetaDataManager {
+
+    @Inject PerWorldInventory plugin;
+    @Inject Injector injector;
+
+    PWIMetaDataManager() {}
+    public <T extends PWIMetaDataValueAbstract> T createMetadataValue(Class<T> clazz, Object... args) {
+        T instance = injector.<T>newInstance(clazz);
+        instance.init(args);
+        return instance;
+    }
+
+    public <T> T getFromPlayer(Player player, String key) {
```

Comment:
Similar to the comment above, it might makes sense here to have two (or three) individual methods for retrieving the values in a typed manner. This way you don't need to pass the `key` from outside—only this class needs to know about the key. 

Compare the current usage:
```java
// I need to pass "lastLocationInWorld" from every class that wants to use it
Map<String, Location> lastLocData = metaDataManager.getFromPlayer(player,"lastLocationInWorld");

// Also, the method is untyped, so nothing stops me from doing this:
Integer lastLocData = metaDataManager.getFromPlayer(player,"lastLocationInWorld"); // compiles!
```

vs. if a method `Map<String, Location> getLastLocationData(Player player)` existed:
```java
Map<String, Location> lastLocData = metaDataManager.getLastLocationData(player);
```